parser/src/test/java/com/facebook/coresql/warning/TestWarningCollector.java

```diff
@@ -0,0 +1,54 @@
+/*
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package com.facebook.coresql.warning;
+
+import com.facebook.coresql.lint.LintingVisitor;
+import com.facebook.coresql.lint.MixedAndOr;
+import com.facebook.coresql.parser.AstNode;
+import org.testng.annotations.Test;
+
+import static com.facebook.coresql.parser.ParserHelper.parseStatement;
+import static org.testng.Assert.assertEquals;
+
+public class TestWarningCollector
+{
+    private static AstNode parse(String sql)
+    {
+        return parseStatement(sql);
+    }
+
+    private static final String hasTwoWarnings = "SELECT a FROM T WHERE a.id = 2 or a.id = 3 and a.age = 73;";
+    public static final String hasOneWarning = "SELECT true or false and false;";
+    private static final LintingVisitor lintingVisitor = new MixedAndOr(new DefaultWarningCollector(new WarningCollectorConfig().setMaxWarnings(1)));
+
+    private static void assertHasOneWarning(String statement)
+    {
+        AstNode ast = parse(statement);
```

Comment:
It's always best to limit the scope of the test to exactly the class being tested. In this case, I don't think it's necessary to parse or lint to test WarningCollector's behavior.

Reply:
The warningCollect.add() method requires an AstNode, which can only be constructed through parsing (it has a protected constructor). But linting isn't necessary. So, I think we need to keep the parsing but can remove the linting.